plugins/woocommerce/includes/gateways/paypal/includes/class-wc-gateway-paypal-helper.php

```diff
@@ -152,4 +152,64 @@ public static function mask_email( $email ) {
 
 		return $masked_local . '@' . $domain;
 	}
+
+	/**
+	 * Update the addresses in the order.
+	 *
+	 * @param WC_Order|null $order The order object.
+	 * @param array         $paypal_order_details The PayPal order details.
+	 * @return void
+	 */
+	public static function update_addresses_in_order( ?WC_Order $order, array $paypal_order_details ): void {
+		if ( empty( $order ) || empty( $paypal_order_details ) ) {
+			return;
+		}
+
+		// Bail early if the addresses update already have been successful.
+		$addresses_update_status = $order->get_meta( '_paypal_address_update_status', true );
+		if ( 'success' === $addresses_update_status ) {
```

Comment:
If we have just 2 statuses for this, I would consider converting it to a boolean flag

Reply:
Done.